src/PureScript/Backend/Chez/Convert.purs

```diff
@@ -191,15 +143,28 @@ codegenExpr codegenEnv@{ currentModule } (NeutralExpr s) = case s of
   EffectDefer _ ->
     S.Identifier "effect-defer"
 
-  PrimOp _ ->
-    S.Identifier "prim-op"
+  PrimOp o ->
+    codegenPrimOp codegenEnv o
   PrimEffect _ ->
     S.Identifier "prim-effect"
   PrimUndefined ->
     S.Identifier "prim-undefined"
 
-  Fail _ ->
-    S.Identifier "fail"
+  Fail i ->
+    -- Note: This can be improved by using `error`, but it requires
+    -- that we track where exactly this `Fail` is defined. We can
+    -- make use of the `codegenEnv` for this.
+    S.List
+      [ S.Identifier "scm:raise"
+      , S.List
+          [ S.Identifier "scm:condition"
+          , S.List [ S.Identifier "scm:make-error" ]
+          , S.List
+              [ S.Identifier "scm:make-message-condition"
+              , S.String $ Json.stringify $ Json.fromString i
```

Comment:
Searching for `Fail` (https://github.com/search?q=repo%3Aaristanetworks%2Fpurescript-backend-optimizer+Fail&type=code), Nate may have anticipated a future need where a more descriptive error message was desired per the `unsafeCrashWith "Unsupported fail."` case in the 7th file from the top of that search.

Reply:
At one point I was, but I think this is only used for pattern match failure. I will probably remove the string the in the future.